src/common/conn/server.rs

```diff
@@ -345,22 +345,29 @@ where
                 },
                 async {
                     let mut api = sig.await.map_err(|e| ServerError::Other(Box::new(e)))?;
-                    if let Some(task) = self.webtrc_conn.as_ref() {
-                        if !task.is_finished() {
-                            log::info!(
-                                "a webrtc connection is active ignoring further signaling requests"
-                            );
-                            return Err(ServerError::ServerConnectionNotConfigured);
-                        }
-                    }
+
+                    let prio = self
+                        .webtrc_conn
+                        .as_ref()
+                        .and_then(|f| (!f.is_finished()).then_some(&current_prio))
+                        .unwrap_or(&None);
+
                     let sdp = api
-                        .answer()
+                        .answer(prio)
                         .await
                         .map_err(|e| ServerError::Other(Box::new(e)))?;
 
+                    if let Some(task) = self.webtrc_conn.take() {
```

Comment:
It took me a moment to see the flow of control here, that if `answer` doesn't fail, it means we didn't reject the new connection based on priority and that we need to cancel (and await cancellation of) the current task. A brief comment before this block would probably go a long way for the next reader.